### DISCUSSION TITLE
Time stepper computed zero time step size on initial which is not allowed.

### DISCUSSION THREAD

**Discussion by jmeier** [idaholab/moose#27350] [Q&A General] [answered]
Dear Moose-Community,

In one of my Moose input files I defined two TimeSequenceSteppers and get the following error:

```
*** ERROR ***
The following error occurred in the object "Executioner", of type "Transient".

Time stepper computed zero time step size on initial which is not allowed.
1. If you are using an existing time stepper, double check the values in your input file or report an error.
2. If you are developing a new time stepper, make sure that initial time step size in your code is computed correctly.
```

My Executioner-block looks like:
```
[Executioner]
  type = Transient

  end_time = 1.0
  [TimeSteppers]
    [TimeSequenceStepper1]
      type = TimeSequenceStepper
      time_sequence = '0.5'
    []
    [TimeSequenceStepper2]
      type = TimeSequenceStepper
      time_sequence = '0.0 0.2 0.4 1.0'
    []
  []
```

I would have expected, that Moose is merging the time steps into '0.0 0.2 0.4 0.5 1.0' rather than giving this error message. My impression from [TimeStepper](https://mooseframework.inl.gov/syntax/Executioner/TimeStepper/) is that Moose should be merging. If I manually merge the two TimeSequenceStepper blocks, Moose works as expected. Is this intended behavior?

**Comment by GiudGiud** [comment, marked as answer]
Hello

> Is this intended behavior?

The error is the intended behavior. Specifying a solve at t=0 when you start at t=0 is disallowed.
The timesequencestepper was ignoring it somehow, but the combined time stepper sees it. 
The combined time stepper does not merge the sequences. It polls the steppers for the next time step to take

Guillaume
